common.mk

```diff
@@ -1936,17 +1936,30 @@ rewindable:
 
 HELP_EXTRA_TASKS = ""
 
+MMTK_BUILD=debug
+MMTK_SRC_PATH=$(srcdir)/gc/mmtk
+MMTK_LIB_PATH=$(srcdir)/gc/mmtk/target/$(MMTK_BUILD)/libmmtk_ruby.$(LIBEXT)
+
 shared-gc: probes.h
 	$(Q) if test -z $(shared_gc_dir); then \
 		echo "You must configure with --with-shared-gc to use shared GC"; \
 		exit 1; \
 	elif test -z $(SHARED_GC); then \
 		echo "You must specify SHARED_GC with the GC to build"; \
 		exit 1; \
+	else \
+		$(MAKEDIRS) $(shared_gc_dir); \
+		echo generating $(shared_gc_dir)librubygc.$(SHARED_GC).$(SOEXT); \
+		if [ "$(SHARED_GC)" = "mmtk" ]; then \
+			if test ! -f $(MMTK_LIB_PATH); then \
+				echo "libmmtk_ruby.$(LIBEXT) not found. Please run 'cargo build' inside $(MMTK_SRC_PATH)"; \
+				exit 1; \
+			fi; \
+			$(LDSHARED) -I$(srcdir)/include -I$(srcdir) -I$(arch_hdrdir) $(XDLDFLAGS) $(cflags) -DBUILDING_SHARED_GC -fPIC -o $(shared_gc_dir)librubygc.$(SHARED_GC).$(SOEXT) $(srcdir)/gc/$(SHARED_GC).c $(MMTK_LIB_PATH); \
+		else \
+			$(LDSHARED) -I$(srcdir)/include -I$(srcdir) -I$(arch_hdrdir) $(XDLDFLAGS) $(cflags) -DBUILDING_SHARED_GC -fPIC -o $(shared_gc_dir)librubygc.$(SHARED_GC).$(SOEXT) $(srcdir)/gc/$(SHARED_GC).c; \
+		fi; \
 	fi
-	$(ECHO) generating $(shared_gc_dir)librubygc.$(SHARED_GC).$(SOEXT)
-	$(Q) $(MAKEDIRS) $(shared_gc_dir)
-	$(Q) $(LDSHARED) -I$(srcdir)/include -I$(srcdir) -I$(arch_hdrdir) $(XDLDFLAGS) $(CFLAGS) $(CPPFLAGS) -DBUILDING_SHARED_GC -fPIC -o $(shared_gc_dir)librubygc.$(SHARED_GC).$(SOEXT) $(srcdir)/gc/$(SHARED_GC).c
```

Comment:
```suggestion
shared-gc: probes.h $(SHARED_GC_LIBRARY)
	$(Q) if test -z $(shared_gc_dir); then \
		echo "You must configure with --with-shared-gc to use shared GC"; \
		exit 1; \
	elif test -z $(SHARED_GC); then \
		echo "You must specify SHARED_GC with the GC to build"; \
		exit 1; \
	fi
	$(ECHO) generating $(shared_gc_dir)librubygc.$(SHARED_GC).$(SOEXT)
	$(Q) $(MAKEDIRS) $(shared_gc_dir)
	$(Q) $(LDSHARED) -I$(srcdir)/include -I$(srcdir) -I$(arch_hdrdir) $(XDLDFLAGS) $(CFLAGS) $(CPPFLAGS) -DBUILDING_SHARED_GC -fPIC -o $(shared_gc_dir)librubygc.$(SHARED_GC).$(SOEXT) $(srcdir)/gc/$(SHARED_GC).c $(SHARED_GC_LIBRARY)
```

Reply:
Sorry, I don't understand what this suggested change is. This just reverts back to the `shared-gc` target on the master branch.

Reply:
The proposal patch is too specific to MMTK.
My suggestion is to separate the needed library as `$(SHARED_GC_LIBRARY)`, and pass it in the command line of `make`.
Building the library is the responsibility of callers.

Reply:
I tried to create a `gc/mmtk/Makefile`, but since I don't know the path of the build directory, I don't have access to the autoconf results in that Makefile, so I don't even know what compiler to use. Do you have an alternative solution or know how to solve this?